src/net.cpp

```diff
@@ -1583,6 +1591,27 @@ size_t V2Transport::GetSendMemoryUsage() const noexcept
     return sizeof(m_send_buffer) + memusage::DynamicUsage(m_send_buffer);
 }
 
+Transport::Info V2Transport::GetInfo() const noexcept
+{
+    AssertLockNotHeld(m_recv_mutex);
+    LOCK(m_recv_mutex);
+    if (m_recv_state == RecvState::V1) return m_v1_fallback.GetInfo();
+
+    Transport::Info info;
+
+    // Do not report v2 and session ID until the garbage authentication packet has been received
+    // and verified (confirming that the other side very likely has the same keys as us).
+    if (m_recv_state != RecvState::KEY_MAYBE_V1 && m_recv_state != RecvState::KEY &&
+        m_recv_state != RecvState::GARB_GARBTERM) {
```

Comment:
I've intentionally tested connecting to a node running a version that still uses the garbage authentication packet (i.e. it doesn't have to recent BIP324 change included yet: https://github.com/bitcoin/bitcoin/pull/28525) and noticed that we report v2 and the session id too early, before the garbage authentication (now included in VERSION) is received. I think that's fixed by the following patch:
```suggestion
        m_recv_state != RecvState::GARB_GARBTERM && m_recv_state != RecvState::VERSION) {
```